process/coordinator/process.go

```diff
@@ -487,38 +484,30 @@ func (tc *transactionCoordinator) CreateMbsAndProcessTransactionsFromMe(
 func (tc *transactionCoordinator) processAddedInterimTransactions() block.MiniBlockSlice {
 	miniBlocks := make(block.MiniBlockSlice, 0)
 
-	tc.mutInterimProcessors.RLock()
-
-	resMutex := sync.Mutex{}
-	// TODO: think if it is good in parallel or it is needed in sequences
-	wg := sync.WaitGroup{}
-	wg.Add(len(tc.interimProcessors))
+	// processing has to be done in order, as the order of different type of transactions over the same account is strict
+	for _, blockType := range tc.keysInterimProcs {
```

Comment:
Might think to guard by mutex tc.keysInterimProcs. Might refactor this by adding a function that will give a pointer-list copy of the available interim procs and iterate freely on it. Valid also for preprocessors

Reply:
made function to get a copy of the keys, which is protected by mutex, than the iterating is more easy to understand. For the pointer-list, every element would be of blockType and the actual preprocessor, a lot of duplicated data.

Reply:
Keys are not used in concurrent setting. ProcessBlockTransaction and CreateBlock are never called by 2 processes at the same time.

Reply:
Ok, since the keys are built only on the constructor we might leave it as it is